data-processor/src/main/java/io/micronaut/data/processor/mappers/jpa/IndexAnnotationMapper.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.processor.mappers.jpa;
+
+
+import io.micronaut.core.annotation.AnnotationValue;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.data.annotation.Index;
+import io.micronaut.inject.annotation.NamedAnnotationMapper;
+import io.micronaut.inject.visitor.VisitorContext;
+
+import java.lang.annotation.Annotation;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Maps JPA's {@code Index} annotation to Micronaut's.
+ *
+ * @author Davide Pugliese
+ * @since 2.4
+ */
+public class IndexAnnotationMapper implements NamedAnnotationMapper {
```

Comment:
I'm not sure if this mapper is actually useful, if I'm correct `javax.persistence.Index` can only be present in the other annotation

Reply:
@dstepanov Now it is for sure as I am using it in the TableAnnotationMapper.